platform-core-module/src/main/java/com/softicar/platform/core/module/program/execution/ProgramExecutionInserter.java

```diff
@@ -27,16 +26,14 @@ public AGProgramExecution insert() {
 
 	private AGProgramExecution insertAndUpdateCurrentExecution() {
 
-		try (var transaction = new DbTransaction()) {
-			program.reloadForUpdate();
-			program//
-				.setCurrentExecution(insertExecution(program.getQueuedBy()))
-				.setQueuedBy(null)
-				.setQueuedAt(null)
-				.save();
-			transaction.commit();
-			return program.getCurrentExecution();
-		}
+		program.reloadProgramStateForUpdate();
```

Comment:
I think this should be rewritten as:
```
var state = program.getState();
state.reloadForUpdate();
state//
    .setCurrentExecution(...)
    .set...
    .save();
```
...so that it's more obvious that this is all about the `state` record.

This, ofc, assumes that `reloadProgramStateForUpdate` was eliminated/inlined.

Reply:
I would like to keep method `reloadStateForUpdate()` because of delegation principle

Reply:
I'd hold the KISS principle against this - but anyway... it's probably just a matter of taste.